### PR TITLE
[MM-20031] Add unit tests to "userDeactivateCmd" to use testify

### PR DESCRIPTION
**Pull request by phillipahereza**
This PR adds unit tests to the `userDeactivateCmd` command

Ticket: [Jira Ticket](https://mattermost.atlassian.net/browse/MM-20031)

Fixes https://github.com/mattermost/mattermost-server/issues/13047

### CODE REVIEW

File: commands/user_test.go [hunk 1]
```diff
@@ -52,3 +52,284 @@ func (s *MmctlUnitTestSuite) TestSearchUserCmd() {
...
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetUserByUsername(argsDelete[2], "").
+			Return(&argUsers[2], &model.Response{Error: nil}).
+			Times(1)
+
+		for i := 0; i < len(argsDelete); i++ {
```
Comment (mgdelacroix) [inline commands/user_test.go]: nit: you can iterate over `argsUsers` to improve readability:

```suggestion
		for _, user := range argUsers {
```

File: commands/user_test.go [hunk 2]
```diff
@@ -52,3 +52,284 @@ func (s *MmctlUnitTestSuite) TestSearchUserCmd() {
...
+
+		s.client.
+			EXPECT().
+			GetUserByEmail(emailArg, "").
+			Return(&mockUser, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			DeleteUser(mockUser.Id).
```
Comment (marianunez) [inline commands/user_test.go]: Just curious, why do we call `DeleteUser` before calling the `userDeactivateCommand`?

**Comment by jasonblais** [comment]
Hey @phillipahereza, can you help with Maria's question above? Seems that's the only outstanding question on this PR. Thanks for your contribution!

File: commands/user_test.go (hunk 2 above)
Comment (phillipahereza) [reply commands/user_test.go]: Sorry for the late response but I assumed the `userDeactivateCommand` is a wrapper around `DeleteUser` so we don't call it before calling `userDeactivateCommand` but rather during the `userDeactivateCommand`.

**Pull request merged**